modules/dnn/src/dnn_utils.cpp

```diff
@@ -382,6 +382,80 @@ void imagesFromBlob(const cv::Mat& blob_, OutputArrayOfArrays images_)
     }
 }
 
+Rect blobRectToImage(Rect r, InputArray oriImage, Image2BlobParams param)
+{
+    CV_Assert(!oriImage.getMat_().empty());
```

Comment:
Instead of accessing to `Mat` you can use [sizend](https://docs.opencv.org/5.x/d4/d32/classcv_1_1__InputArray.html#a7cb4cadf3108dc232e05e3ae5550e912) for a direct access to the shape